gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/CommitActivityImpl.java

```diff
@@ -75,11 +74,14 @@ public int commit(WUProcessingSpec workSpec) {
       Path jobOutputPath = new Path(new Path(jobIdParent, "output"), jobIdParent.getName());
       log.info("Output path at: " + jobOutputPath + " with fs at " + fs.getUri());
       StateStore<TaskState> taskStateStore = Help.openTaskStateStore(workSpec, fs);
-      Collection<TaskState> taskStateQueue =
-          ImmutableList.copyOf(
-              TaskStateCollectorService.deserializeTaskStatesFromFolder(taskStateStore, jobOutputPath, numDeserializationThreads));
-      commitTaskStates(jobState, taskStateQueue, globalGobblinContext);
-      return taskStateQueue.size();
+      Optional<Queue<TaskState>> taskStateQueue =
+              TaskStateCollectorService.deserializeTaskStatesFromFolder(taskStateStore, jobOutputPath.getName(), numDeserializationThreads);
+      if (!taskStateQueue.isPresent()) {
+        log.error("No task states found at " + jobOutputPath);
+        return 0;
+      }
+      commitTaskStates(jobState, ImmutableList.copyOf(taskStateQueue.get()), globalGobblinContext);
```

Comment:
same comment as above w/ `.get()` only once, then no further use of the `Optional`